ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -847,11 +886,14 @@ private PutChunk getFreeChunk() {
   private PutChunk getBuildingChunk() {
     PutChunk chunkToReturn = null;
     for (PutChunk chunk : putChunks) {
-      if (chunk.isBuilding()) {
+      if (chunk.isBuilding() && (firstPutChunkPrepared || chunk != firstChunk)) {
```

Comment:
I don't think we want to treat first chunk differently here with building. If the first chunk is currently building, let's just return it.